test/Generator/PropertyGeneratorTest.php

```diff
@@ -387,14 +388,14 @@ public function testFromReflectionOmitsDefaultValueIfItIsNull(): void
         $this->assertSame('    public static $fooStaticProperty;', $code);
     }
 
-    public function testFromReflectionOmitsTypeHintInTypedProperty(): void
+    public function testFromReflectionWithTypeHintInTypedProperty(): void
```

Comment:
please don't change existing tests to avoid regression, add new one instead.

Reply:
This was changed on purpose of adding 'type hinting' to 'TypedProperty'. Without modifying this, the test would definitely fail. Please advise what I should do.

Reply:
except `readonly` which must has type hint, this new feature should be optional to avoid regression.

Reply:
You may need to add `$omitTypeHint` 2nd parameter to `PropertyGenerator::fromReflection` method, with default true. 

@Ocramius any opinion about this?

Reply:
I double checked  the composer.json and seems the minimum PHP version requirement is 7.4, which has type hint  for property already. Am I right to say we don't have to omit  the generation of type hint for backward compatibility at all? Hence the need for changing the test is necessary, because it seems to me that the original test was purely created because the lack of ability to generate type hint which should have been done a long time ago.

Reply:
after making the changes (change $type from string to TypeGenerator)as per requested by @Ocramius 
I think the only conflict we have here is whether we should modify the existing unit test, which was objected by @samsonasik.

Let me re-clarify the reason I changed the existing unit test:
1. The old test naming was a bit misleading. There was no option to 'omit' type hint, because there was no way to generate type hint on property until my PR.
2. The old test was created in https://github.com/laminas/laminas-code/commit/0872dbda5d3a8b680a6af4dae4fe76f7e0c9714a , and in the original PR https://github.com/laminas/laminas-code/pull/77 the author explained clearly: 'The test explicitly checks, that typed properties will be ignored when generating the code. Maybe that feature should be implemented in a future release, and that will cause this test to fail.'

So it's pretty obvious to me such change to the existing unit test is necessary / mandatory.